subsys/usb/class/hid/core.c

```diff
@@ -68,8 +68,8 @@ USBD_CLASS_DESCR_DEFINE(primary) struct usb_hid_config hid_cfg = {
 
 static void usb_set_hid_report_size(u16_t report_desc_size)
 {
-	hid_cfg.if0_hid.subdesc[0].wDescriptorLength =
-		sys_cpu_to_le16(report_desc_size);
+	UNALIGNED_PUT(sys_cpu_to_le16(report_desc_size),
+		      &hid_cfg.if0_hid.subdesc[0].wDescriptorLength);
 }
```

Comment:
I wonder could we use here sys_put_le16()

Reply:
I see it now, should work, @SavinayDharmappa can you try it?